src/main/java/org/zalando/nakadi/service/subscription/StreamParameters.java

```diff
@@ -83,7 +83,7 @@ public boolean isStreamLimitReached(final long commitedEvents) {
     }
 
     public boolean isKeepAliveLimitReached(final IntStream keepAlive) {
-        return batchKeepAliveIterations.map(it -> keepAlive.allMatch(v -> v >= it)).orElse(false);
+        return batchKeepAliveIterations.map(it -> keepAlive.allMatch(v -> v >= it) && it > 0).orElse(false);
```

Comment:
@m99coder thank you for contribution. 
I think the better place to check that condition would be `StreamParameters` constructor https://github.com/zalando/nakadi/blob/67e95a474593ef785357edaac509689e88901a22/src/main/java/org/zalando/nakadi/service/subscription/StreamParameters.java#L64 as it's a place where we handle all those special values of stream parameters and default values.